experiment/build/gcb_build.py

```diff
@@ -52,17 +54,32 @@ def build_coverage(benchmark):
         '_GCS_COVERAGE_BINARIES_DIR': coverage_binaries_dir,
         '_BENCHMARK': benchmark,
     }
-    config_file = get_build_config_file('coverage.yaml')
+    config = generate_cloudbuild.generate_coverage_images_build_spec()
     config_name = 'benchmark-{benchmark}-coverage'.format(benchmark=benchmark)
-    _build(config_file, config_name, substitutions)
+    _build(config, config_name, substitutions)
 
 
-def _build(config_file: str,
+def build_fuzzer_benchmark(fuzzer: str, benchmark: str) -> bool:
+    """Builds |benchmark| for |fuzzer|."""
+    substitutions = {
+        '_BENCHMARK': benchmark,
+        '_FUZZER': fuzzer,
+    }
+    config = generate_cloudbuild.generate_benchmark_images_build_spec()
+    config_name = 'benchmark-{benchmark}-fuzzer-{fuzzer}'.format(
+        benchmark=benchmark, fuzzer=fuzzer)
+
+    _build(config, config_name, substitutions)
+
+
+def _build(config: Dict[str, str],
            config_name: str,
            substitutions: Dict[str, str] = None,
            timeout_seconds: int = GCB_BUILD_TIMEOUT
           ) -> new_process.ProcessResult:
     """Build each of |args| on gcb."""
+    config_file = os.path.join(utils.ROOT_DIR, 'docker', 'cloudbuild.yaml')
```

Comment:
OK now I see it. We are still doing variable GCB's variable substitution.
So basically we generate this file for base-images. Then when we build coverage images we overwrite it each time we do a build but we overwrite it with the same thing so it's fine. Then when we do fuzzer-benchmark builds we do the same thing but we overwrite it a bunch of times with the same file as well.
Is my understanding correct?
If so, I think this is pretty fragile. We should fix this by using tempfile.

Reply:
I did use tempfile in the beginning, but the read always failed in the actual execution. I can look more into that.

Reply:
Did you use namedtemporary file? Are you sure the file wasn't deleted before it was used?

Reply:
Yes, I used named and the file wasn't deleted because I set the delete param to not delete it on close as well.

Reply:
Did you keep variable substituion because this didn't work?

Reply:
No, the two were independent. I planned substitution in that case as well, because the way its getting `${_FUZZER}` and the benchmark substitution is by using a common method from `docker_images.py`, from where it uses format to place these variables instead of `{fuzzer}`.